Aplib.Core/ExposedQueue.cs

```diff
@@ -0,0 +1,193 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// A queue with all elements exposed.
+    /// Functionally works like a queue with indexing.
+    /// </summary>
+    public class ExposedQueue<T> : ICollection<T>
+    {
+        /// <summary>
+        /// The length of the array.
+        /// </summary>
+        public int MaxCount { get; private set; }
+
+        /// <summary>
+        /// Actual number of elements in the array.
+        /// Limited to <see cref="MaxCount"/>.
+        /// </summary>
+        public int Count
+        {
+            get => _count;
+            private set => _count = value > MaxCount ? MaxCount : value;
```

Comment:
I'd say the count would never exceed MaxCount? If that is the case, and it gets caught here, we did something wrong. If the value is higher than MaxCount, an exception should be thrown instead of returning the max value.

Reply:
I did it this way to avoid having to put `if (Count < MaxCount) Count++` in the `Put()` method. I can put it there, but then I would not put another check for an exception, since that would never occur.